src/storage/statistics_cl.c

```diff
@@ -508,3 +518,255 @@ stats_dump (const char *class_name_p, FILE * file_p)
 
   fprintf (file_p, "\n\n");
 }
+
+/*
+ * stats_ndv_dump () - Dumps the NDV about a class
+ *   return:
+ *   classname(in): The name of class to be printed
+ *   fp(in):
+ */
+void
+stats_ndv_dump (const char *class_name_p, FILE * file_p)
+{
+  MOP class_mop;
+  int error = NO_ERROR;
+  DB_ATTRIBUTE *att;
+  int col_cnt = 0;
+  CLASS_ATTR_NDV class_attr_ndv;
+  int i;
+
+  class_mop = sm_find_class (class_name_p);
+  if (class_mop == NULL)
+    {
+      return;
+    }
+
+  /* get NDV by query */
+  if (stats_get_ndv_by_query (class_mop, &class_attr_ndv, file_p) != NO_ERROR)
+    {
+      goto end;
+    }
+
+  /* print NDV info */
+  fprintf (file_p, "\nNumber of Distinct Values\n");
+  fprintf (file_p, "****************\n");
+  fprintf (file_p, " Class name: %s\n", class_name_p);
+  for (i = 0; i < class_attr_ndv.attr_cnt; i++)
+    {
+      fprintf (file_p, "  %s (%ld)\n", sm_get_att_name (class_mop, class_attr_ndv.attr_ndv[i].id),
+	       class_attr_ndv.attr_ndv[i].ndv);
+    }
+  fprintf (file_p, "total count : %ld\n", class_attr_ndv.attr_ndv[i].ndv);
+  fprintf (file_p, "\n");
+
+end:
+  if (class_attr_ndv.attr_ndv != NULL)
+    {
+      free_and_init (class_attr_ndv.attr_ndv);
+    }
+  return;
+}
+
+/*
+ * stats_get_ndv_by_query () - get NDV by query
+ *   return:
+ *   class_mop(in):
+ *   attr_ndv(out):
+ */
+int
+stats_get_ndv_by_query (const MOP class_mop, CLASS_ATTR_NDV * class_attr_ndv, FILE * file_p)
+{
+  DB_ATTRIBUTE *att;
+  int error = NO_ERROR;
+  char *select_list = NULL;
+  const char *query = NULL;
+  char *query_buf = NULL;
+  int buf_size = 0;
+  DB_QUERY_RESULT *query_result = NULL;
+  DB_QUERY_ERROR query_error;
+  DB_VALUE value;
+  INT64 v1, v2;
+  DB_DOMAIN *dom;
+  ATTR_NDV *att_ndv = NULL;
+  int i;
+  const char *class_name_p = NULL;
+
+  /* get class_name */
+  class_name_p = db_get_class_name (class_mop);
+  /* count number of the columns */
+  class_attr_ndv->attr_cnt = 0;
+  att = (DB_ATTRIBUTE *) db_get_attributes_force (class_mop);
+  while (att != NULL)
+    {
+      class_attr_ndv->attr_cnt++;
+      att = db_attribute_next (att);
+    }
+
+  class_attr_ndv->attr_ndv = (ATTR_NDV *) malloc (sizeof (ATTR_NDV) * (class_attr_ndv->attr_cnt + 1));
+  if (class_attr_ndv->attr_ndv == NULL)
+    {
+      return ER_FAILED;
+    }
+
+  select_list = stats_make_select_list_for_ndv (class_mop, &class_attr_ndv->attr_ndv);
+  if (select_list == NULL)
+    {
+      return ER_FAILED;
+    }
+
+  /* create sampling SQL statement */
+  query = "SELECT /*+ SAMPLING_SCAN */ %s FROM [%s]";
+
+  buf_size = strlen (select_list) + strlen (class_name_p) + 40;
+  query_buf = (char *) malloc (sizeof (char) * buf_size);
+  if (query_buf == NULL)
+    {
+      return ER_FAILED;
+    }
+  snprintf (query_buf, buf_size, query, select_list, class_name_p);
```

Comment:
Identifier names must be enclosed in square brackets.

```
csql> create table t1 ([year] int);
csql> ;info ndv t1
Query : SELECT /*+ SAMPLING_SCAN */ count(distinct year), count(*) FROM [dba.t1]

Committed.
csql> ;ex

$ cat csql.err

Time: 05/02/23 18:35:20.537 - SYNTAX ERROR *** file src/parser/query_result.c, line 305 ERROR CODE = -493, Tran = 1, EID = 5
Syntax: invalid use of year
 YEAR( expression )

Time: 05/02/23 18:35:20.537 - SYNTAX ERROR *** file src/compat/db_vdb.c, line 994 ERROR CODE = -492, Tran = 1, EID = 6
invalid use of year
 YEAR( expression )
```

Reply:
Thank you. I fixed it.